beacon_node/http_api/src/lib.rs

```diff
@@ -1933,6 +1933,51 @@ pub fn serve<T: BeaconChainTypes>(
             })
         });
 
+    /*
+     * builder/states
+     */
+
+    let builder_states_path = eth_v1
+        .and(warp::path("builder"))
+        .and(warp::path("states"))
+        .and(chain_filter.clone());
+
+    // GET builder/states/{state_id}/expected_withdrawals
+    let get_expected_withdrawals = builder_states_path
+        .clone()
+        .and(warp::path::param::<StateId>())
+        .and(warp::path("expected_withdrawals"))
+        .and(warp::path::end())
+        .and_then(|chain: Arc<BeaconChain<T>>, state_id: StateId| {
+            blocking_json_task(move || {
+                let (state, execution_optimistic, finalized) = match state_id.state(&chain) {
+                    Ok(state) => state,
+                    Err(_) => {
+                        return Err(warp_utils::reject::custom_not_found(
+                            "State not found".to_string(),
+                        ))
+                    }
+                };
+                let forkchoice_update_parameters = chain
+                    .canonical_head
+                    .cached_head()
+                    .forkchoice_update_parameters();
+                match chain.get_expected_withdrawals(&forkchoice_update_parameters, state.slot()) {
+                    Ok(withdrawals) => Ok(api_types::GenericResponse::from(
+                        api_types::NextWithdrawalsResponse {
+                            data: withdrawals,
+                            execution_optimistic: execution_optimistic,
+                            finalized: finalized,
```

Comment:
Clippy lints will error here - the redundant field names should be removed. You can run `make lint` to run lint locally.

```suggestion
                            execution_optimistic,
                            finalized,
```